File: macos/QMK Toolbox/USB/Bootloader/HalfKayDevice.m
#import "HalfKayDevice.h"

@implementation HalfKayDevice

- (id)initWithUSBDevice:(USBDevice *)usbDevice {
    if (self = [super initWithUSBDevice:usbDevice]) {
        self.name = @"HalfKay";
        self.type = BootloaderTypeHalfKay;
        self.resettable = YES;
    }
    return self;
}

-(void)flashWithMCU:(NSString *)mcu file:(NSString *)file {
    [self runProcess:@"teensy_loader_cli" withArgs:@[[NSString stringWithFormat:@"-mmcu=%@", mcu], file, @"-v"]];
}

-(void)resetWithMCU:(NSString *)mcu {
    [self runProcess:@"teensy_loader_cli" withArgs:@[[NSString stringWithFormat:@"-mmcu=%@", mcu], @"-bv"]];
}

@end

File: macos/QMK Toolbox/USB/Bootloader/HalfKayDevice.swift
import Foundation

class HalfKayDevice: BootloaderDevice {
    override init(usbDevice: USBDevice) {
        super.init(usbDevice: usbDevice)
        name = "HalfKay"
        type = .halfKay
        resettable = true
    }

    override func flash(_ mcu: String, file: String) {
        runProcess("teensy_loader_cli", args: ["-mmcu=\(mcu)", file, "-v"])
    }

    override func reset(_ mcu: String) {
        runProcess("teensy_loader_cli", args: ["-mmcu=\(mcu)", "-bv"])
    }
}
